execd.go

```diff
@@ -312,7 +312,9 @@ func handleChannel(conn *ssh.ServerConn, newChan ssh.NewChannel, execHandler []s
 				}
 				cmd = exec.Command(execHandler[0], append(execHandler[1:], cmdargs...)...)
 			}
-			if !*env {
+			if *env {
+				cmd.Env = os.Environ()
+			} else {
 				cmd.Env = []string{}
```

Comment:
Yeah, this is really not doing anything to the default value. And because we next almost always append values to it, it is no longer nil when the command is executed so it doesn't use the parent env.